test/validate-checkconfig.sh

```diff
@@ -18,6 +18,15 @@ for org in openshift redhat-operator-ecosystem; do
   # We need to enter the git directory and run git commands from there, our git
   # is too old to know the `-C` option.
   pushd "${clonedir}/${org}"
+
+  # First we'll run registry-replacer to prune unused images.
+  registry-replacer \
```

Comment:
Does this only perform the image pruning?  It seems to have added a non-trivial amount of time to the test:

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_ci-tools/2127/pull-ci-openshift-ci-tools-master-checkconfig/1413572707332657152#1:build-log.txt%3A11

vs.

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_ci-tools/2138/pull-ci-openshift-ci-tools-master-checkconfig/1410710606075400192#1:build-log.txt%3A11

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_ci-tools/2164/pull-ci-openshift-ci-tools-master-checkconfig/1414979492593537024#1:build-log.txt%3A11

Reply:
It also does some image tag replacement stuff. Part of it is reading the Dockerfile from the GH repo, which would be network IO, but that is currently a pre-requisite to doing the image pruning as well.

Reply:
> that is currently a pre-requisite to doing the image pruning

In what way?  The only case we'd miss by pruning without first replacing is if the code happened to remove a base image added by the replacement process (and if that was a unique case that was not already present in `release`), correct?

I think we can get enough signal by just pruning `release` as is without performing the expensive replacement in this test.

Reply:
I changed it around so that the base image pruning is not dependent upon the dockerfile image replacement stuff. Also, I added a flag to allow bypassing the image replacement that can be used in this test. Let me know if that makes sense to you.